libcurlcpp.pc.in

```diff
@@ -0,0 +1,11 @@
+prefix="@CMAKE_INSTALL_PREFIX@"
+libdir="${prefix}"
```

Comment:
This seems... wrong? libdir is not the same as prefix, libdir is usually, but not always, `${prefix}/lib`

In the absence of functionality such as Meson's `join_paths('${prefix}', libdir_opt)` you probably also want to use https://github.com/jtojnar/cmake-snips/blob/master/CMakeScripts/JoinPaths.cmake

Reply:
Fixed in the latest draft.

Reply:
But then it still doesn't respect CMAKE_INSTALL_LIBDIR.

Reply:
I pointed to the JoinPaths.cmake script for this. Just using CMAKE_INSTALL_FULL_LIBDIR is *also* wrong, because it means that pkg-config doesn't know what the prefix is and cannot do sysroot things or relocation.

Reply:
Thanks, it appears properly fixed now, all that is left is testing. I spent hours on this on Friday but your advice made me figure it out in minutes.

Reply:
Like I just said, it is still not properly fixed. Just using ` Libs: -L@CMAKE_INSTALL_FULL_LIBDIR@ -lcurlcpp` produces incorrect results too, just in different circumstances. `${prefix}` exists for a reason.

Reply:
Does it look right now?
Here's a generated one in which I did set CMAKE_INSTALL_PREFIX manually:
```Text
prefix=/home/jan/Work/contrib/curlcpp/install
libdir=${prefix}/include
includedir=${prefix}/lib

Name: curlcpp
Description: An object oriented C++ wrapper for CURL (libcurl)
URL: https://github.com/JosephP91/curlcpp/
Version: 1.4
Requires.private: libcurl >= 7.28.0
Cflags: -I${libdir}
Libs: -L${includedir} -lcurlcpp
```

Reply:
```
 libdir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
```

CMake supports setting e.g. `-DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_LIBDIR=/usr/lib` and simply joining them together with `/` can then fail and produce `/usr/usr/lib`.

I keep saying this, and I have an actual reason for it, but... please.

Use JoinPaths.cmake

Reply:
Yes, all this would be significantly easier if cmake would just include a builtin pkg-config generator that consistently does the right thing and doesn't require thousands of people to rediscover the same problems again and again. Like the one Meson includes (``pkgconfig.generate(lib)`).

Reply:
Thank you. I'm reading the README.md of the JoinPaths.cmake repository, I'll push another commit when I'm done.